.github/actions/setup-opensearch-dashboards/action.yml

```diff
@@ -0,0 +1,269 @@
+      if: ${{ inputs.install_zip == 'true' && runner.os == 'Linux' }}
+      with:
+        url: https://ci.opensearch.org/ci/dbc/distribution-build-opensearch-dashboards/${{ steps.osd-version.outputs.plugin-version }}/latest/linux/x64/tar/builds/opensearch-dashboards/dist/opensearch-dashboards-min-${{ steps.osd-version.outputs.plugin-version }}-linux-x64.tar.gz
+
+    # Extract downloaded zip
+    - name: Extract downloaded tar
+      if: ${{ inputs.install_zip == 'true' && runner.os == 'Linux' }}
+      run: |
+        tar -xzf opensearch-*.tar.gz
+        rm -f opensearch-*.tar.gz
+      shell: bash
+
+    - id: determine-dashboards-directory-zip-install
+      if: ${{ inputs.install_zip == 'true' }}
+      run: echo "dashboards-directory=opensearch-dashboards-${{ steps.osd-version.outputs.plugin-version }}-linux-x64"  >> $GITHUB_OUTPUT
+      shell: bash
+
+    - name: Build the plugin 
+      if: ${{ inputs.install_zip == 'true' }}
+      run: |
+          yarn build
+      working-directory: ${{ steps.determine-plugin-directory.outputs.plugin-directory }}
+      shell: bash
+
+    - name: Install the plugin zip
+      if: ${{ inputs.install_zip == 'true' && inputs.built_plugin_suffix == '' }}
+      run: | 
+        ${{ steps.determine-dashboards-directory-zip-install.outputs.dashboards-directory }}/bin/opensearch-dashboards-plugin install file:$(pwd)/OpenSearch-Dashboards/plugins/${{ inputs.plugin_name }}/build/${{ inputs.built_plugin_name }}-${{env.PLUGIN_VERSION}}.zip
+      shell: bash
+
+    - name: Install the plugin zip
+      if: ${{ inputs.install_zip == 'true' &&  inputs.built_plugin_suffix != '' }}
+      run: | 
+        ${{ steps.determine-dashboards-directory-zip-install.outputs.dashboards-directory }}/bin/opensearch-dashboards-plugin install file:$(pwd)/OpenSearch-Dashboards/plugins/${{ inputs.plugin_name }}/build/${{ inputs.built_plugin_name }}-${{inputs.built_plugin_suffix}}.zip
+      shell: bash
+
+    - name: Replace opensearch_dashboards.yml file if applicable binary
+      if: ${{ inputs.opensearch_dashboards_yml != '' && inputs.install_zip == 'true' }}
+      run: |
+          mv ${{ inputs.opensearch_dashboards_yml }} ${{ steps.determine-dashboards-directory-zip-install.outputs.dashboards-directory }}/config/opensearch_dashboards.yml
+      shell: bash
+
+    - name: Replace opensearch_dashboards.yml file if applicable dev
+      if: ${{ inputs.opensearch_dashboards_yml != '' && inputs.install_zip != 'true' }}
+      run: |
+          mv ${{ inputs.opensearch_dashboards_yml }} ./OpenSearch-Dashboards/config/opensearch_dashboards.yml
+      shell: bash
+
+    - name: Add to opensearch_dashboards.yml config file
+      run: |
+        echo 'opensearch_security.configuration.admin_pages_enabled: false' >> ./OpenSearch-Dashboards/config/opensearch_dashboards.yml
+        echo 'opensearch_security_admin.configuration.session_management_enabled: false' >> ./OpenSearch-Dashboards/config/opensearch_dashboards.yml
```

Comment:
For my knowledge, I do have some conceptual questions about these flags:
- what are these flags for? (I guess it is for disable/enable the 2 security dashboards we installed, but why we disabled both of them here?)
- Are they enabled by default?

Reply:
Notice that the prefixes are different `opensearch_security` vs `opensearch_security_admin`.

For one of the plugin installations it disables the Admin pages and only manages the session, including login/logout, changing tenant and viewing user info. For the other plugin, it disables session management. In that mode, the plugin acts like any other dashboard plugin where it displays pages for the user to interact with security. It takes for granted that another plugin manages the session.